controllers/openstackdataplane_controller.go

```diff
@@ -137,11 +137,11 @@ func (r *OpenStackDataPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 
 	var deployErrors []string
 	shouldRequeue := false
+	roles := &dataplanev1.OpenStackDataPlaneRoleList{}
```

Comment:
roles will only be populated if deploy=true. I think you need to move the the  r.Client.List for the roles to be outside the deploy "if".

Reply:
nice catch!